### PR TITLE
new manage_requirements argument to address issue 273

### PR DESCRIPTION
**Pull request by rosenbergj**

### CODE REVIEW

**Comment by rosenbergj** [comment]
(My first ever pr; please be gentle...)

**Comment by ghoneycutt** [comment]
@rosenbergj thanks for contributing!

File: manifests/requirements.pp [hunk 1]
```diff
@@ -38,6 +38,9 @@
 # [*extra_pip_args*]
 # Extra arguments to pass to pip after the requirements file
 #
+# [*manage_requirements*]
```
Comment (ghoneycutt) [inline manifests/requirements.pp]: This also needs to be documented in the README.md

File: manifests/requirements.pp [hunk 2]
```diff
@@ -110,7 +114,7 @@
 
   # This will ensure multiple python::virtualenv definitions can share the
   # the same requirements file.
-  if !defined(File[$requirements]) {
+  if !defined(File[$requirements]) and $manage_requirements {
```
Comment (ghoneycutt) [inline manifests/requirements.pp]: Add the `== true` bit makes it easier to understand exactly what is happening.

```puppet
if !defined(File[$requirements]) and $manage_requirements == true {
```

File: manifests/requirements.pp (hunk 1 above)
Comment (rosenbergj) [inline manifests/requirements.pp]: Fixed.

File: manifests/requirements.pp (hunk 2 above)
Comment (rosenbergj) [inline manifests/requirements.pp]: Good idea. Done.

**Comment by ghoneycutt** [comment]
Looks good. We also need some spec tests - See https://github.com/stankevich/puppet-python/blob/master/spec/defines/requirements_spec.rb

Add a block where you let params set manage_requirements to false and it should test that the file `$requirements` is not in the catalog

``` ruby
context "/requirements.txt" do
        let (:params) {{ :requirements => "/requirements.txt", :manage_requirements => false }}
        it { is_expected.not_to contain_file("/requirements.txt") }
      end
```

**Comment by rosenbergj** [comment]
Thank you!

**Pull request merged**